File: backend/app/models/work_entry.py
from sqlalchemy import Column, Integer, String, Numeric, Boolean, DateTime, Date, Text, ForeignKey, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base
import enum


class WorkStatus(str, enum.Enum):
    completed = "completed"
    pending = "pending"
    cancelled = "cancelled"


class WorkEntry(Base):
    __tablename__ = "work_entries"

    id = Column(Integer, primary_key=True, index=True)
    client_id = Column(Integer, ForeignKey("clients.id"), nullable=False)
    category_id = Column(Integer, ForeignKey("categories.id"), nullable=False)
    title = Column(String(255), nullable=False)
    work_date = Column(Date, nullable=False)
    rate = Column(Numeric(10, 2), nullable=False)  # Snapshot rate at time of entry
    notes = Column(Text, nullable=True)
    status = Column(
        Enum(WorkStatus),
        nullable=False,
        default=WorkStatus.completed
    )
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    client = relationship("Client", back_populates="work_entries")
    category = relationship("Category", back_populates="work_entries")

File: backend/src/models/WorkEntry.js
import mongoose from 'mongoose';

const workEntrySchema = new mongoose.Schema({
  client_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Client', required: true },
  category_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Category', required: true },
  title: { type: String, required: true, trim: true },
  work_date: { type: Date, required: true },
  rate: { type: Number, required: true, min: 0 },
  notes: { type: String, default: null },
  status: { type: String, enum: ['completed', 'pending', 'cancelled'], default: 'completed' },
}, { timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }, versionKey: false });

export default mongoose.model('WorkEntry', workEntrySchema);
